setup.py

```diff
@@ -110,7 +110,6 @@ def get_extras_require() -> Dict[str, List[str]]:
             "dask-ml",
             "botorch ; python_version>'3.6'",
             "fastai",
-            "jax",
```

Comment:
`jax` is required to execute `haiku_simple.py` since it is an extra requirement of `dm-haiku`.

https://github.com/deepmind/dm-haiku/blob/b7348274c24d1faa94c1348d5e9301c0e806b62e/setup.py#L38-L41

Actually, `dm-haiku-0.0.3` only installs `absl-py-0.11.0`,  `numpy-1.20.0` `six-1.15.0`, and the example failed due to the following error.

```console
python examples/haiku_simple.py
Traceback (most recent call last):
  File "examples/haiku_simple.py", line 17, in <module>
    import haiku as hk
  File "/usr/local/lib/python3.7/site-packages/haiku/__init__.py", line 17, in <module>
    from haiku import data_structures
  File "/usr/local/lib/python3.7/site-packages/haiku/data_structures.py", line 17, in <module>
    from haiku._src.data_structures import to_immutable_dict
  File "/usr/local/lib/python3.7/site-packages/haiku/_src/data_structures.py", line 29, in <module>
    from haiku._src import utils
  File "/usr/local/lib/python3.7/site-packages/haiku/_src/utils.py", line 23, in <module>
    import jax
ModuleNotFoundError: No module named 'jax'
```

Reply:
I'm really sorry but I didn't uninstall `jax` correctly in the verification... I reproduced the error.
As you pointed out, `jax` is in `extra_requires`, `jax` isn't installed unless `[jax]` is specified in `pip install`.

Reply:
Yes, it works. I can install `jax` via installing of `optax`, not `dm-haiku`.

https://github.com/deepmind/optax/blob/master/setup.py#L47

Reply:
The commands I ran are as folllows:

```bash
pip uninstall jax optax dm-haiku
pip install -e .[example]
```

Reply:
Thank you for your information. I'll check it.

Reply:
I appreciate your double-check :)

Reply:
I confirmed that it worked with the following commands:

```console
$ docker run -it -v $(pwd):/prj --rm -w /prj python:3.7 bash
$ pip install -e .
$ apt-get update
$ apt-get -y install openmpi-bin libopenmpi-dev
$ pip install -e ."[example]" -f https://download.pytorch.org/whl/torch_stable.html
$ python examples/haiku_simple.py
```

Thank you!